crates/edr_provider/src/requests/methods.rs

```diff
@@ -134,8 +134,7 @@ pub enum MethodInvocation {
     #[serde(rename = "eth_getStorageAt")]
     GetStorageAt(
         Address,
-        /// position
-        U256,
+        #[serde(deserialize_with = "crate::requests::eth::deserialize_storage_index")] U256,
```

Comment:
My mental model is that `edr_eth::serde` is for deserialization of data according to conventions in the Ethereum ecosystem, while `deserialize_storage_index` is input validation for the `eth_getStorageAt` handler, that's why I put it next to `handle_get_storage_at_request`. Happy to change it though

Reply:
My understanding is that this is explicitly part of the Ethereum spec, both based on our discussion with Pato and the execution spec: https://ethereum.github.io/execution-apis/api-documentation/ (they require a unsigned int of exactly 32 bytes)

```rust
title hex encoded 256 bit unsigned integer
type string
pattern ^0x([1-9a-f]+[0-9a-f]{0,31})|0$
```

So based on your own logic and for coherency it would make most send to put it in `serde`

Reply:
Resolving as we discussed on Slack and decided that as the error messages we return from  `deserialize_storage_index` are specific to Hardhat, it's ok to not to follow `edr_eth::serde` pattern.